blockmanager.go

```diff
@@ -3043,4 +3085,188 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
 	return newLightHeaderCtx(
 		ancestorHeight, ancestor, l.store, l.headerList,
 	)
+
+}
+
+func (b *blockManager) sideLoadHeaders() {
+
+	// If sideloading is not enabled on this chain
+	// return quickly.
+	if !b.cfg.sideLoad.Enabled {
+		return
+	}
+
+	// If headers contained in the side load source are for a different chain network return
+	// immediately.
+	if b.sideLoadReader.HeadersChain() != b.cfg.ChainParams.Net {
+
+		log.Error("headers from side load file are of network %v "+
+			"and so incompatible with neutrino's current bitcoin network -- skipping side loading", b.sideLoadReader.HeadersChain())
+
+		return
+	}
+	log.Infof("Side loading headers from %v to %v", b.headerTip, b.sideLoadReader.EndHeight())
+
+	// Set headerTip to enable reader supply header, node needs
+	err := b.sideLoadReader.SetHeight(int64(b.headerTip))
+
+	if err != nil {
+		log.Errorf("error while setting height for sideload--- skipping sideloading: "+
+			"%v", err)
+
+		return
+	}
+	headerBatch := make([]headerfs.BlockHeader, 0)
+	for {
+		//Request header
+		header, headerErr := b.sideLoadReader.NextHeader()
+		// If any error occurs while fetching headers that does not indicate
+		// an end of file, return immediately.
+		if headerErr != nil && headerErr != io.EOF {
```

Comment:
`NextHeaders(n int64)` could return a custom err for when there are not more headers that we catch here to break the `for`

Reply:
I would implement the custom error but I think we should leave it to fetching a header at a time because of the mode of verification. So we skip creating another loop inside a loop to verify.